### PR TITLE
chore(web): add dataSource, layerSidePanel component in beta

### PR DESCRIPTION
**Pull request by pyshx**
# Overview
This PR adds supports for layers in ReEarth Beta, with main focus on DataSource and LayerSidePanel Component.

### CODE REVIEW

File: web/src/beta/components/Icon/Icons/addLayer.svg
Comment (KaWaite) [inline web/src/beta/components/Icon/Icons/addLayer.svg:1]: With any new svg, please remove fills, etc and replace with `currentColor`. If you don't, we can't customize the color of the icon through styled components

File: web/src/beta/components/Modal/index.tsx [hunk 1]
```diff
@@ -79,11 +82,13 @@ const Modal: React.FC<Props> = ({
         )}
         <ContentWrapper>
           {tabs.length > 0 && <Content>{tabs.find(tab => tab.active === true)?.content}</Content>}
-          <Content> {children}</Content>
-          <ButtonWrapper>
-            {tabs.find(tab => tab.active === true)?.tabButton1 ?? button1}
-            {tabs.find(tab => tab.active === true)?.tabButton2 ?? button2}
-          </ButtonWrapper>
+          {children && <Content> {children}</Content>}
```
Comment (KaWaite) [inline web/src/beta/components/Modal/index.tsx:85]: Double check your IDE is working with our linting? I see a gap between <Content> and {children}

File: web/src/beta/components/SidePanelSectionField/index.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ const SidePanelSectionField: React.FC<{
 };
 
 const Field = styled.div`
-  background: ${({ theme }) => theme.bg[1]};
+  background: ${({ theme }) => theme.bg[0]};
```
Comment (KaWaite) [inline web/src/beta/components/SidePanelSectionField/index.tsx:31]: This messes up the background of other components that use this. Please revert this and wrap SidePanelSectionField as a styled component to change the background color 🙏🏻

File: web/src/beta/features/Editor/DataSourceManager/index.tsx [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+`;
+
+const ColJustiftBetween = styled.div`
+  display: flex;
+  height: 100%;
+  flex-direction: column;
+  justify-content: space-between;
+`;
+
+const DelimitedText: React.FC<DataProps> = ({ sceneId, onSubmit, onClose }) => {
```
Comment (KaWaite) [inline web/src/beta/features/Editor/DataSourceManager/index.tsx:188]: Please split all these components in to their own files 🙏🏻

File: web/src/beta/features/Editor/index.tsx [hunk 4]
```diff
@@ -45,16 +58,27 @@ const Editor: React.FC<Props> = ({ sceneId, projectId, workspaceId, tab, stories
     stories,
   });
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { selectedLayer, handleLayerAdd, handleLayerDelete, handleLayerSelect, handleLayerUpdate } =
```
Comment (KaWaite) [inline web/src/beta/features/Editor/index.tsx:62]: You shouldn't destruture anything that you don't need

File: web/src/beta/features/Editor/DataSourceManager/asset.tsx
Comment (KaWaite) [inline web/src/beta/features/Editor/DataSourceManager/asset.tsx:1]: In react there is the convention of naming files for components with uppercase. Please update all names with uppercase 🙏🏻

File: web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import { t } from "i18next";
+
+import SidePanelSectionField from "@reearth/beta/components/SidePanelSectionField";
+import SidePanelCommon from "@reearth/beta/features/Editor/SidePanel";
+import type { NLSLayer } from "@reearth/services/api/layersApi/utils";
+
+import Layers from ".";
+
+function LeftPanel() {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (KaWaite) [inline web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.stories.tsx:11]: This shouldn't be needed.

File: web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  const groups = [
+    {
+      schemaGroupId: "scene",
+      title: "Scene",
+    },
+    {
+      schemaGroupId: "layers",
+      title: "Layers",
```
Comment (KaWaite) [inline web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.stories.tsx:19]: Please wrap the title in i18n

File: web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+import SidePanelSectionField from "@reearth/beta/components/SidePanelSectionField";
+import SidePanelCommon from "@reearth/beta/features/Editor/SidePanel";
+import type { NLSLayer } from "@reearth/services/api/layersApi/utils";
+
+import Layers from ".";
+
+function LeftPanel() {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const groups = [
```
Comment (KaWaite) [inline web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.stories.tsx:12]: useMemo

File: web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+      schemaGroupId: "scene",
+      title: "Scene",
+    },
+    {
+      schemaGroupId: "layers",
+      title: "Layers",
+    },
+  ];
+
+  const layers: NLSLayer[] = [
```
Comment (KaWaite) [inline web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.stories.tsx:23]: useMemo

File: web/src/services/api/sceneApi.ts [hunk 9]
```diff
@@ -23,6 +23,7 @@ export type SceneExtension = {
 export type SceneQueryProps = {
   sceneId?: string;
   lang?: string;
+  pollInterval?: number;
```
Comment (KaWaite) [inline web/src/services/api/sceneApi.ts:26]: Is this being used?

Comment (pyshx) [reply web/src/services/api/sceneApi.ts:26]: Its not, https://www.apollographql.com/docs/react/data/queries/#polling

File: web/src/beta/features/Editor/hooks.ts [hunk 10]
```diff
@@ -56,7 +57,12 @@ export default ({ tab }: { tab: Tab }) => {
     [setWidgetEditor],
   );
 
+  const { useGetLayersQuery } = useLayersFetcher();
+
+  const { nlsLayers = [] } = useGetLayersQuery({ sceneId, pollInterval: 500 });
```
Comment (KaWaite) [inline web/src/beta/features/Editor/hooks.ts:62]: Why do you not do this in the useLayers.ts file? Seems appropriate to go in there, no?

File: web/src/beta/features/Editor/index.tsx [hunk 11]
```diff
@@ -22,15 +27,23 @@ type Props = {
 };
 
 const Editor: React.FC<Props> = ({ sceneId, projectId, workspaceId, tab, stories }) => {
+  const showDataSourceManager = useReactiveVar(showDataSourceManagerVar);
```
Comment (KaWaite) [inline web/src/beta/features/Editor/index.tsx:30]: I don't know we need this to be global state, do you? If you think so, lets discuss quickly on zoom, since I am currently just doing local state and passing down as props similar state that if there is a good reason to make global maybe i should move those up too.
If not, please just do useState here and then pass down where necessary

File: web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+              onClick: () => onLayerDelete(l.id),
+            },
+          ]}
+        />
+      ))}
+    </StyledLayerContainer>
+  );
+};
+
+// utils
```
Comment (KaWaite) [inline web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx:74]: Not sure what this means, but either way remove it. The name of the component should be enough without a comment

File: web/src/beta/features/Editor/useLayers.ts [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+};
+
+export default function ({ nlsLayers }: Props) {
+  const t = useT();
+  const { useAddNLSLayerSimple, useRemoveNLSLayer, useUpdateNLSLayer } = useLayersFetcher();
+  const [selectedLayerId, setSelectedLayerId] = useState<string | undefined>(undefined);
+
+  const selectedLayer = useMemo(() => {
+    return nlsLayers.length ? nlsLayers[0] : undefined;
+  }, [nlsLayers]);
```
Comment (KaWaite) [inline web/src/beta/features/Editor/useLayers.ts:34]: Shouldn't selected layer be the layer corresponding with the selectedLayerId??

File: web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.stories.tsx (hunk 5 above)
Comment (pyshx) [reply web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.stories.tsx]: Done.

File: web/src/beta/features/Editor/hooks.ts (hunk 10 above)
Comment (pyshx) [reply web/src/beta/features/Editor/hooks.ts]: Done.

File: web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx (hunk 12 above)
Comment (pyshx) [reply web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx]: Done.

File: web/src/beta/features/Editor/useLayers.ts (hunk 13 above)
Comment (pyshx) [reply web/src/beta/features/Editor/useLayers.ts]: Done.

File: web/src/beta/features/Editor/index.tsx (hunk 11 above)
Comment (pyshx) [reply web/src/beta/features/Editor/index.tsx:30]: I'll explain on zoom.

File: web/src/services/state/index.ts [hunk 14]
```diff
@@ -9,8 +10,21 @@ export type SelectedWidget = {
...
 export const selectedWidgetVar = makeVar<SelectedWidget | undefined>(undefined);
 
 export const selectedWidgetAreaVar = makeVar<WidgetAreaType | undefined>(undefined);
 
+export const selectedLayerVar = makeVar<SelectedLayer | undefined>(undefined);
+
+export const showDataSourceManagerVar = makeVar<boolean>(false);
+
+export const showPopOverLayerButtonVar = makeVar<boolean>(true);
+
```
Comment (KaWaite) [inline web/src/services/state/index.ts:29]: I've said this elsewhere about the showDataSouceManagerVar, but these maybe don't need to be global, so please move them more locally and pass down as props 🙏🏻

File: web/src/beta/features/Editor/DataSourceManager/DelimtedText/index.tsx
Comment (KaWaite) [inline web/src/beta/features/Editor/DataSourceManager/DelimtedText/index.tsx:1]: Folder name type: DelimtedText -> DelimitedText

File: web/src/services/api/sceneApi.ts (hunk 9 above)
Comment (KaWaite) [reply web/src/services/api/sceneApi.ts:26]: Ahh sorry yeah can you for now remove the poll. You can leave the typings as is, just remove where you are passing the actual poll value. 
The reason is because I am worried about performance issues around using polling, and want to investigate just how far apollo client's polling gets us with "real-time" working together on the same project.
If it is a good solution, I want to setup polling on most queries. But also want to investigate other ways to keep synced up.

Comment (pyshx) [reply web/src/services/api/sceneApi.ts:26]: Done.

File: web/src/services/state/index.ts (hunk 14 above)
Comment (pyshx) [reply web/src/services/state/index.ts]: Done.

File: web/src/beta/features/Editor/DataSourceManager/DelimtedText/index.tsx
Comment (pyshx) [reply web/src/beta/features/Editor/DataSourceManager/DelimtedText/index.tsx:1]: Done.

File: web/codegen.ts [hunk 15]
```diff
@@ -25,6 +25,7 @@ const config: CodegenConfig = {
           URL: "string",
           Lang: "string",
           TranslatedString: "{ [lang in string]?: string } | null",
+          JSON: "any",
```
Comment (KaWaite) [inline web/codegen.ts:28]: Why is this change necessary?

Comment (pyshx) [reply web/codegen.ts:28]: We've added a new Scalar named `JSON` in the backend for couple of things. Its better to put it here.

**Review by KaWaite — APPROVED** [review]

**Review by nina992 — APPROVED** [review]
